### PR TITLE
OBJLoader Update: Quick bandages to support/fix the new way that

### PR DESCRIPTION
**Pull request by shadekiller666**
… normals are calculated, a very quick bandage to patch TextureCoordinates for the time being, and the Parser now uses a Pattern to split strings on whitespace instead of only splitting on " ".

### CODE REVIEW

**Comment by shadekiller666** [comment]
Oh, and an additional change to MultiModel that was causing MultiModel.Baked to fire a "RuntimeException: No base model or submodel found for this MultiModel.Baked." when trying to bake the missing block model.

File: src/main/java/net/minecraftforge/client/model/obj/OBJLoader.java [hunk 1]
```diff
@@ -68,13 +68,24 @@ else if (modelLocation.getResourcePath().startsWith("models/item/"))
...
                 }
                 OBJModel.Parser parser = new OBJModel.Parser(resource, manager);
-                OBJModel model = parser.parse();
-                cache.put(modelLocation, model);
+                OBJModel model = null;
+                try
+                {
+                	model = parser.parse();
+                }
+                catch (OBJModel.UVsOutOfBoundsException e)
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/client/model/obj/OBJLoader.java]: This catch block is completely unnecessary, omit it and just leave the finally block

File: src/main/java/net/minecraftforge/client/model/obj/OBJModel.java [hunk 2]
```diff
@@ -368,23 +378,32 @@ else if (key.equalsIgnoreCase("g") || key.equalsIgnoreCase("o"))
...
             }
-            OBJModel model = new OBJModel(this.materialLibrary, this.objFrom);
-            model.getMatLib().setUVBounds(minU, maxU, minV, maxV);
+            
+            OBJModel model;
+            if (this.UVsOutOfBounds)
+            {
+            	throw new UVsOutOfBoundsException(this.objFrom);
+            }
+            else
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/client/model/obj/OBJModel.java]: No else required here

File: src/main/java/net/minecraftforge/client/model/obj/OBJModel.java [hunk 3]
```diff
@@ -1406,36 +1411,32 @@ else if (this.state instanceof TRSRTransformation)
...
                 switch (format.getElement(e).getUsage())
                 {
                     case POSITION:
-                        builder.put(e, v.getPosition().x, v.getPosition().y, v.getPosition().z, v.getPosition().w);
+                        builder.put(e, v.getPos().x, v.getPos().y, v.getPos().z, v.getPos().w);
                         break;
                     case COLOR:
-                        float d = LightUtil.diffuseLight(n.normal.x, n.normal.y, n.normal.z);
+                        float d;
+                        if (v.hasNormal()) d = LightUtil.diffuseLight(v.getNormal().x, v.getNormal().y, v.getNormal().z);
```
Comment (tterrag1098) [inline src/main/java/net/minecraftforge/client/model/obj/OBJModel.java]: Pretty sure these sameline ifelse's go against forge style. Plus they are unreadable. No one is counting lines of code, use braces.

**Pull request merged**